pandas/core/dtypes/inference.py

```diff
@@ -283,11 +285,13 @@ def is_list_like(obj):
     False
     """
 
-    return (isinstance(obj, compat.Iterable) and
+    return (isinstance(obj, compat.Iterable)
             # we do not count strings/unicode/bytes as list-like
-            not isinstance(obj, string_and_binary_types) and
+            and not isinstance(obj, string_and_binary_types)
             # exclude zero-dimensional numpy arrays, effectively scalars
-            not (isinstance(obj, np.ndarray) and obj.ndim == 0))
+            and not (isinstance(obj, np.ndarray) and obj.ndim == 0)
+            # exclude sets if allow_sets is False
```

Comment:
blank line before comments

Reply:
style nit: I don't like the blank lines inside an if condition.

But nothing that needs to be changed now.